### ISSUE TITLE
Header enum value is not getting generated as available values on @fastify/swagger

### ISSUE THREAD

**Issue by vishnutmohan** [fastify/fastify-swagger#701]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Plugin version

6.1.1

### Node.js version

14.17

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.0.1

### Description

I have defined my schema for the headers like below.

I have defined my schema for the headers like below.
```
const headers = {
  $id: "headers",
  type: "object",
  properties: {
    Authorization: { type: "string" },
    "x-country": {
      type: "string",
      transform: ["toUpperCase"],
      enum: ["IN", "US", "CH", "AR", "MX", "UY", "BR"]
    },
    "x-commerce": {
      type: "string",
      description: "Example values: 'FA', 'SO', 'TO' etc"
    }
  }
};
```

And added in one of my request header like
```
exports.createProductSizeSchema = {
  tags: ["Product Size"],
  summary: "This API is used to create  Product Sizes",
  description: "productSize should be Unique",
  headers: { $ref: "headers#" }
}
```

But while generating the definition, enum value is not coming up. PFB the screenshot
<img width="570" alt="Screenshot 2022-12-05 at 8 39 57 PM" src="https://user-images.githubusercontent.com/1974570/205672174-3a9eeaa3-0b8e-46fa-92f1-f07f1bd65e57.png">

Could you please let me know what I am doing wrong here?

### Steps to Reproduce

* Create a fastify instance and on registering plugins

```
const swagger = require("@fastify/swagger");

fastify.register(swagger, {
  routePrefix: "/documentation",
  swagger: {
    info: {
      title: "Test API",
      description: "API Docs",
      version: "0.1.0"
    },
    servers: [
    ],
    externalDocs: {
      url: "https://swagger.io",
      description: "Find more info here"
    },
    consumes: ["application/json"],
    produces: ["application/json"]
  },
  exposeRoute: true
});
```

### Expected Behavior

Expecting a response like
<img width="729" alt="Screenshot 2022-12-06 at 10 46 39 AM" src="https://user-images.githubusercontent.com/1974570/205822758-dca4cab6-1747-4188-97d8-799159271b30.png">

**Comment by mcollina** [comment]
@climba03003 could you take a look?

**Comment by climba03003** [comment]
The fix should be simple enough to allow more key for `header`.

https://github.com/fastify/fastify-swagger/blob/e11df8c11fb4c46fd8c58dced115509af0b33eac/lib/spec/openapi/utils.js#L208-L216

**Comment by WoH** [comment]
Should be fixed in 8.3.0 via #704 .
